DataFormats/SoATemplate/interface/SoACommon.h

```diff
@@ -0,0 +1,693 @@
+#ifndef DataFormats_SoATemplate_interface_SoACommon_h
+#define DataFormats_SoATemplate_interface_SoACommon_h
+
+/*
+ * Definitions of SoA common parameters for SoA class generators
+ */
+
+#include <cstdint>
+#include <cassert>
+#include <ostream>
+#include <tuple>
+#include <type_traits>
+
+#include <boost/preprocessor.hpp>
+
+#include "FWCore/Utilities/interface/typedefs.h"
```

Comment:
It's there because we were supposed to use `cms_int32_t`... I'm not sure when it got lost, but I've added it back, together with a BuildFile.xml.

Yes, `boost_header` should be the one with the minimal dependency in boost (not sure why it depends on `sockets`, but that uses only system libraries anyway).